src/PKSim.Core/Services/SimulationParametersToBuildingBlockUpdater.cs

```diff
@@ -51,23 +67,73 @@ public ICommand UpdateParametersFromSimulationInBuildingBlock(Simulation simulat
-         updateCommands.BuildingBlockType = buildingBlockType;
-         updateCommands.BuildingBlockName = templateBuildingBlock.Name;
-         updateCommands.CommandType = PKSimConstants.Command.CommandTypeUpdate;
-         updateCommands.Description = PKSimConstants.Command.UpdateTemplateBuildingBlockCommandDescription(buildingBlockType, templateBuildingBlock.Name, simulation.Name);
-         _executionContext.UpdateBuildingBlockPropertiesInCommand(updateCommands, templateBuildingBlock);
-
+       
          _executionContext.PublishEvent(new BuildingBlockUpdatedEvent(templateBuildingBlock));
          return updateCommands;
       }
 
-      private IOSPSuiteCommand updateParameterValues(PathCache<IParameter> simulationParameters, PathCache<IParameter> templateParameters)
+      private ICommand synchronizeBuildingBlocks(IPKSimBuildingBlock templateBuildingBlock, IPKSimMacroCommand updateTemplateParametersCommand)
+      {
+         var simulationSubject = templateBuildingBlock as ISimulationSubject;
+         //For now, deal with update from Individual or Population into Expression Profile
+         if (simulationSubject == null)
+            return new PKSimEmptyCommand();
+
+         var allExpressionProfileParameterValueCommand = updateTemplateParametersCommand.All()
+            .OfType<IExpressionProfileCommand>()
+            .OfType<EditParameterCommand>()
+            .ToList();
+
+         if (!allExpressionProfileParameterValueCommand.Any())
+            return new PKSimEmptyCommand();
+
+
+         var expressionProfilesToUpdate = new HashSet<ExpressionProfile>();
+         var macroCommand = new PKSimMacroCommand();
+         //We have some commands related to expression profile. We need to update the expression profile
+         foreach (var parameterCommand in allExpressionProfileParameterValueCommand)
+         {
+            var simulationSubjectParameter = _executionContext.Get<IParameter>(parameterCommand.ParameterId);
+            if (simulationSubjectParameter == null)
+               continue;
+
+            //This should be the id of the parameter in the expression profile
+            var expressionProfileParameterId = simulationSubjectParameter.Origin.ParameterId;
+            var expressionProfileParameter = _executionContext.Get<IParameter>(expressionProfileParameterId);
+            if (expressionProfileParameter == null)
+               continue;
+
+            var expressionProfile = _executionContext.BuildingBlockContaining(expressionProfileParameter) as ExpressionProfile;
+            if (expressionProfile == null)
+               continue;
+
+            expressionProfilesToUpdate.Add(expressionProfile);
+
+
+            //We do not update the simulation subject. It will be done at the end of the loop
+            var command =  new SetExpressionProfileValueCommand(expressionProfileParameter, simulationSubjectParameter.Value, updateSimulationSubjects: false);
```

Comment:
setting the flag to false so that it does not take too much time. We will update all at once at the end